pages/api/flows/[id].js

```diff
@@ -1,20 +1,31 @@
 import dbConnect from "../../../lib/dbConnect";
 import Flow from "../../../models/Flow";
 import { getFlowById } from "../../../services/flowService";
+import { unstable_getServerSession } from "next-auth/next";
+import { authOptions } from "../auth/[...nextauth]";
 
 export default async function handler(request, response) {
   const { id } = request.query;
+  const session = await unstable_getServerSession(
+    request,
+    response,
+    authOptions
+  );
   await dbConnect();
 
   if (request.method === "GET") {
     const flow = await getFlowById(id);
     return response.status(201).json(flow);
```

Comment:
Your API route is not protected here. I can enter the following URL and I get the data back which is private and should only be visible to the user: [Link to unprotected API route](https://capstone-project-git-feat-user-login-cathrintruchan.vercel.app/api/flows/636e19113482ba9309f4a2a3)